### PR TITLE
Tbolt/3726 Subform Enhancement

### LINKED ISSUE #3726: Address any UX regressions introduced from the subform updates
In ticket #3646 we updated all of the subforms behavior such that they no longer use the auto-saving feature. This introduces an issue where the frontend and backend can get out of sync. 
If for example you add 3 new milestones but don't save any to the backend:
1. milestone 1
2. milestone 2
3. milestone 3

What is supposed to happen when you save milestone 2? Since this is an ordered list we submit milestone 2 to the backend and the backend adds that to the list at the `1.` slot and the other in-progress milestone forms are discarded. 

The team has come up with some ideas on how to resolve:
* Don't show the "Add new [item]" after one is added
* Show an alert before any milestone form data will be lost

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #3726 

**Description-**

This updates the behavior of the "Add [Subfom name]" buttons. Now when a subform item is added, the option to add additional ones is not available.

**Steps to manually verify this change...**

1. Login with a user that can view/edit APDs
2. Navigate to each subform
3. Press "Add [name of subform]" button
4. Verify that the "Add [name of subform]" is no longer visible
5. Cancel or add the subform item and verify the "Add [name of subform]" is available and visible again

### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: integrationTests/cypress/integration/02-apd/01-apd-basics.spec.js [hunk 1]
```diff
@@ -301,6 +302,7 @@ describe('APD Basics', { tags: ['@apd', '@default'] }, () => {
       cy.get('@primaryContactVals').contains('Edit').should('exist');
 
       cy.findByRole('button', { name: /Add Key Personnel/i }).click();
+      cy.findByRole('button', { name: /Add Key Personnel/i }).should('not.exist');
```
Comment (Sun-Mountain) [inline integrationTests/cypress/integration/02-apd/01-apd-basics.spec.js:305]: Wasn't the plan to disable the button?

**Review by Sun-Mountain — APPROVED** [review]
⭐

**Review by thetif — APPROVED** [review]

**Comment by jeromeleecms** [comment]
What's the expectation when I add a subform and hit save? Should the default be that a user cannot save right after they hit the button? Right now the save button his active and I can hit save (no form validation occurs) and I can save an blank record.  #3980 works as intended (hiding the button), but is the save button part of this or separate?

**Review by mirano-darren — APPROVED** [review]
Well done! Jerome bring up a good point, validation doesn't kick off instantly. Could you initially set the subform to invalid?

**Comment by tbolt** [comment]
@jeromeleecms @mirano-darren Subform in-line validation is current work-in-progress. You can check out #3499 and the other validation PRs to see the behavior. They all disable the Save button until the form is completed though.

**Review by jeromeleecms — APPROVED** [review]
Looks good! Per the comments, validation will be implemented through other issues/pr's.

**Review by beparticular — APPROVED** [review]

**Pull request merged**